llvm/utils/TableGen/Common/GlobalISel/PatternParser.cpp

```diff
@@ -107,13 +107,10 @@ getInstrForIntrinsic(const CodeGenTarget &CGT, const CodeGenIntrinsic *I) {
 static const CodeGenIntrinsic *getCodeGenIntrinsic(Record *R) {
   // Intrinsics need to have a static lifetime because the match table keeps
   // references to CodeGenIntrinsic objects.
-  static DenseMap<const Record *, std::unique_ptr<CodeGenIntrinsic>>
-      AllIntrinsics;
-
-  auto &Ptr = AllIntrinsics[R];
-  if (!Ptr)
-    Ptr = std::make_unique<CodeGenIntrinsic>(R);
-  return Ptr.get();
+  static CodeGenIntrinsicMap *AllIntrinsics;
+  if (!AllIntrinsics)
+    AllIntrinsics = new CodeGenIntrinsicMap(R->getRecords());
+  return &(*AllIntrinsics)[R];
```

Comment:
Why does this need its own special instance of the map?

Reply:
WDYM? This is at the moment just replacing the static DenseMap used in that function. We could actually use a non-static Map, instantiated in `EmitGICombiner` in GlobalISelCombinerEmitter.cpp and pass the reference around as well, but that's beyond intended the scope of this MR. I will do that as a follow on.

Reply:
I mean I would expect there to be one central place that the intrinsic map is created that all backends use, not this one static case in GlobalIsel patterns